### PR TITLE
Moving more stuff out of the MTRDevice

### PR DESCRIPTION
**Pull request by woody-apple**
MTRDevice is still using some basic C++ stuff here, we should be not doing this in MTRDevice_XPC, I did a quick and simple hack here. @bzbarsky-apple

### CODE REVIEW

File: src/darwin/Framework/CHIP/MTRDevice_XPC.mm [hunk 1]
```diff
@@ -309,4 +309,41 @@ - (void)downloadLogOfType:(MTRDiagnosticLogType)type
 // Not Supported via XPC
 // - (oneway void)downloadLogOfType:(MTRDiagnosticLogType)type nodeID:(NSNumber *)nodeID timeout:(NSTimeInterval)timeout completion:(void (^)(NSURL * _Nullable url, NSError * _Nullable error))completion;
 
+#pragma mark - Storage Overrides
+
+- (NSDictionary<MTRClusterPath *, MTRDeviceClusterData *> *)_clusterDataToPersistSnapshot
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRDevice_XPC.mm:314]: If the goal is to not have persistence at all in MTRDevice_XPC, then we should just rip it out of MTRDevice, not do this, I think....

File: src/darwin/Framework/CHIP/MTRDevice_Concrete.mm [hunk 2]
```diff
@@ -1646,11 +1646,6 @@ - (void)_scheduleClusterDataPersistence
     }
     [_mostRecentReportTimes addObject:[NSDate now]];
 
-    {
-        std::lock_guard lock(_descriptionLock);
-        _mostRecentReportTimeForDescription = [_mostRecentReportTimes lastObject];
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRDevice_Concrete.mm:1651]: Why is this being removed?

File: src/darwin/Framework/CHIP/MTRDevice_XPC.mm (hunk 1 above)
Comment (woody-apple) [reply src/darwin/Framework/CHIP/MTRDevice_XPC.mm:314]: Have at it :)

File: src/darwin/Framework/CHIP/MTRDevice_Concrete.mm (hunk 2 above)
Comment (woody-apple) [reply src/darwin/Framework/CHIP/MTRDevice_Concrete.mm:1651]: I just copied and pasted the base implementation into concrete, please fix :)

**Comment by bzbarsky-apple** [comment]
I think we should do https://github.com/project-chip/connectedhomeip/pull/36132 instead.

**Pull request closed without merging**